app/services/file_locator.rb

```diff
@@ -118,7 +118,7 @@ def reader
     when 'file'
       File.open(location,'r')
     else
-      Kernel::open(uri.to_s, 'r')
```

Comment:
`Kernel::open` would fail to find the sharepoint files due to our Ruby version. Using `Kernel::open` to call `URI.open` was removed in Ruby 3.0, so we need to explictly use `URI.open` here.